### DISCUSSION TITLE
Sharing routes

### DISCUSSION THREAD

**Discussion by mfik** [vercel/turborepo#6139] [Help] [answered]
For a project i have two next.js projects. Both follow the same structure for every path in api / app. Is there a possibility to share these routes?

The structure now: 
```
- web1
- - app
- --pages
- ---index.tsx
- web2
- - - app
- --pages
- ---index.tsx
```
Both indexes are the same, as many folders. I would love one codebase, where we can update the pages in a package and the route is still added to the app folders of both websites.

I have thought about a catch all route, checking for a export in a "app-routes" package and displaying <Page /> if available, but i think there might be better ways to this, if available.

Is this possible?

**Comment by anthonyshew** [comment, marked as answer]
This is possible - but it depends how you'd like to accomplish it.

The most straightforward way that comes to mind for me is extracting the page as if it were a simple React component to a package and then importing it into page routes.
